onebusaway-android/src/main/java/org/onebusaway/android/ui/ArrivalInfo.java

```diff
@@ -451,31 +252,28 @@ private String computeTimeLabel(Context context) {
         }
     }
 
-    /**
-     * Computes the arrival status label from the delay (i.e., schedule deviation), where positive
-     * means the bus is running late and negative means the bus is running ahead of schedule
-     *
-     * @param delay schedule deviation, in minutes, for this vehicle where positive
-     *              means the bus is running late and negative means the bus is running ahead of
-     *              schedule
-     * @return the arrival status label based on the deviation
-     */
-    public static String computeArrivalLabelFromDelay(Resources res, long delay) {
-        if (delay > 0) {
-            // Arriving delayed
-            return res.getQuantityString(
-                    R.plurals.stop_info_arrive_delayed, (int) delay,
-                    delay);
-        } else if (delay < 0) {
-            // Arriving early
-            delay = -delay;
-            return res
-                    .getQuantityString(
-                            R.plurals.stop_info_arrive_early,
-                            (int) delay, delay);
+    private String computeNotifyText(Context context) {
+        final String routeName = mInfo.getShortName();
```

Comment:
This should be changed to:

`final String displayRouteName = UIUtils.getRouteDisplayName(mInfo.getRouteLongName(), mInfo.getShortName());`

Related unit tests will also need to be updated.

Only `route_long_name` OR `route_short_name` is required by GTFS, so unfortunately we can't count on `route_short_name` always existing.  I recently hit this issue in https://github.com/OneBusAway/onebusaway-android/issues/409#issuecomment-169457263 with the Seattle ferries, for which `route_short_name` doesn't exist.